src/stacked_borrows.rs

```diff
@@ -80,6 +81,22 @@ impl SbTagExtra {
     }
 }
 
+#[derive(Debug)]
+pub struct FrameExtra {
+    /// The ID of the call this frame corresponds to.
+    pub call_id: CallId,
+
+    /// If this frame is protecting any tags, they are listed here. We use this list to do
+    /// incremental updates of the global list of protected tags stored in the
+    /// `stacked_borrows::GlobalState` upon function return, and if we attempt to pop a protected
+    /// tag, to identify which call is responsible for protecting the tag.
+    /// See `Stack::item_popped` for more explanation.
+    ///
+    /// This will contain one tag per reference passed to the function, so
+    /// a size of 2 is enough for the vast majority of functions.
+    pub protected_tags: SmallVec<[SbTag; 2]>,
```

Comment:
The fields should be private, like the other types defined here. (You'll also have to make `new_call` a constructor for this type; probably best to rename it `new_frame` or so.)

The idea is that we should be able to evolve this type without touching any other file.

Reply:
Ohhhhh now I understand